### PR TITLE
NOTFORMERGE: SLUDGE: Add IT track player

### PR DESCRIPTION
**Pull request by rsf92**

### CODE REVIEW

**Comment by sev-** [comment]
I converted it to a draft.

The intended usage of this PR is to have this library added for usage in the SLUDGE engine that uses ImpulseTracker modules in some games.

File: configure [hunk 1]
```diff
@@ -277,6 +278,7 @@ add_feature gif "GIF" "_gif"
 add_feature theoradec "libtheoradec" "_theoradec"
 add_feature tinygl "TinyGL" "_tinygl"
 add_feature vorbis "Vorbis file support" "_vorbis _tremor"
+add_feature minkom "MOD file support" "_mikmod"
```
Comment (sev-) [inline configure:281]: minkom -> mikmod

**Comment by criezy** [comment]
As a head up, AGS games can also use ImpulseTracker mod music, but this is currently not implemented in ScummVM. So if would be great if the code is added in such a way that it can be used in other engines.

**Comment by rsf92** [comment]
> As a head up, AGS games can also use ImpulseTracker mod music, but this is currently not implemented in ScummVM. So if would be great if the code is added in such a way that it can be used in other engines.

I guess including a wrapper class for the library in https://github.com/scummvm/scummvm/tree/master/audio/decoders would be the best way to go, right?

**Comment by criezy** [comment]
> I guess including a wrapper class for the library in https://github.com/scummvm/scummvm/tree/master/audio/decoders would be the best way to go, right?

I think what would be most logical in term of integration is to implement it as an `AudioStream` in https://github.com/scummvm/scummvm/tree/master/audio/mods (similarly to what is done for ProTracker and S3M/XM).

**Comment by rsf92** [comment]
> > I guess including a wrapper class for the library in https://github.com/scummvm/scummvm/tree/master/audio/decoders would be the best way to go, right?
> 
> I think what would be most logical in term of integration is to implement it as an `AudioStream` in https://github.com/scummvm/scummvm/tree/master/audio/mods (similarly to what is done for ProTracker and S3M/XM).

Fine, I think I had already inherited that class in my current code, but I'm a long way to go yet, this was not the best week in terms of free time

**Comment by sev-** [comment]
I am closing this for the time being. Once there is a real code and progress, it could always be opened in a new PR.

**Pull request closed without merging**